examples/check-expression/check_expression.md

```diff
@@ -1,9 +1,7 @@
 # Check expression
 
-`check E` is used with an expression `E` that might result in an `error` value. If `E` results in an `error` value , then, `check` makes the function return that `error` value immediately.
-
-The type of `check E` does not include `error`. The control flow remains explicit.
+In Ballerina, it is common to write an expression that may result in an error, such as calling a function that could return an error value, checking if the result belongs to the `error` type and immediately returning that value. You can use the `check` expression to simplify this pattern.
```

Comment:
```suggestion
In Ballerina, it is common to write an expression that may result in an error, such as calling a function that could return an error value, checking if the result belongs to the `error` type and immediately returning the value if so. You can use the `check` expression to simplify this pattern.
```

But, it's not just returning the error is it? We explain that later in the check semantics BBE, but wondering if we should use the term fails and say generally returning the error from the function.